diff --git a/dist/adapters/node.mjs b/dist/adapters/node.mjs
index bf860db..845af0b 100644
--- a/dist/adapters/node.mjs
+++ b/dist/adapters/node.mjs
@@ -717,7 +717,7 @@ function toNodeHandler(handler) {
 			req: nodeReq,
 			res: nodeRes
 		}));
-		return res instanceof Promise ? res.then((resolvedRes) => sendNodeResponse(nodeRes, resolvedRes)) : sendNodeResponse(nodeRes, res);
+		return (res && typeof res.then === 'function') ? res.then((resolvedRes) => (resolvedRes && typeof resolvedRes.then === 'function') ? resolvedRes.then((r) => sendNodeResponse(nodeRes, r)) : sendNodeResponse(nodeRes, resolvedRes)) : sendNodeResponse(nodeRes, res);
 	}
 	convertedNodeHandler.__fetchHandler = handler;
 	assignFnName(convertedNodeHandler, handler, " (converted to Node handler)");
@@ -772,7 +772,7 @@ var NodeServer = class {
 			});
 			request.waitUntil = this.#wait?.waitUntil;
 			const res = fetchHandler(request);
-			return res instanceof Promise ? res.then((resolvedRes) => sendNodeResponse(nodeRes, resolvedRes)) : sendNodeResponse(nodeRes, res);
+			return (res && typeof res.then === 'function') ? res.then((resolvedRes) => (resolvedRes && typeof resolvedRes.then === 'function') ? resolvedRes.then((r) => sendNodeResponse(nodeRes, r)) : sendNodeResponse(nodeRes, resolvedRes)) : sendNodeResponse(nodeRes, res);
 		};
 		this.node = {
 			handler,
